Port the remaining mediastream related types to the new IPC serialization format

gavin-apple opened the pull request:
#### c46ed9ed7b256953e3af9ad614a42d90dc08d8ea
<pre>
Port the remaining mediastream related types to the new IPC serialization format
<a href="https://bugs.webkit.org/show_bug.cgi?id=251860">https://bugs.webkit.org/show_bug.cgi?id=251860</a>
rdar://105132585

Reviewed by Youenn Fablet.

This change ports the remaining MediaStream related types. This includes:
    - MediaConstraint::DataType
    - MediaConstraintType
    - MediaConstraint
    - IntConstraint
    - DoubleConstraint
    - BooleanConstraint
    - StringConstraint
    - FrameRateRange
    - VideoPresetData
    - RealtimeMediaSourceSupportedConstraints
    - VideoFacingMode
    - DisplaySurfaceType
    - RealtimeMediaSourceSettings
    - CaptureDevice::DeviceType
    - CaptureDevice
    - RealtimeMediaSourceCapabilities::EchoCancellation
    - RTCDataChannelInit
    - RTCDataChannelIdentifier

* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/platform/mediastream/CaptureDevice.h:
(WebCore::CaptureDevice::encode const): Deleted.
(WebCore::CaptureDevice::decode): Deleted.
* Source/WebCore/platform/mediastream/MediaConstraints.h:
(WebCore::NumericConstraint::NumericConstraint):
(WebCore::StringConstraint::StringConstraint):
(WebCore::MediaConstraint::encode const): Deleted.
(WebCore::MediaConstraint::decode): Deleted.
(WebCore::NumericConstraint::encode const): Deleted.
(WebCore::NumericConstraint::decode): Deleted.
(WebCore::StringConstraint::encode const): Deleted.
(WebCore::StringConstraint::decode): Deleted.
* Source/WebCore/platform/mediastream/RTCDataChannelHandler.h:
(WebCore::RTCDataChannelInit::encode const): Deleted.
(WebCore::RTCDataChannelInit::decode): Deleted.
* Source/WebCore/platform/mediastream/RTCDataChannelIdentifier.h:
(): Deleted.
(WebCore::RTCDataChannelIdentifier::encode const): Deleted.
(WebCore::RTCDataChannelIdentifier::decode): Deleted.
* Source/WebCore/platform/mediastream/RTCDataChannelLocalIdentifier.h: Copied from Source/WebCore/platform/mediastream/RTCDataChannelIdentifier.h.
* Source/WebCore/platform/mediastream/RealtimeMediaSource.h:
* Source/WebCore/platform/mediastream/RealtimeMediaSourceCapabilities.h:
(WebCore::RealtimeMediaSourceCapabilities::RealtimeMediaSourceCapabilities):
(WebCore::RealtimeMediaSourceCapabilities::facingMode const):
(WebCore::RealtimeMediaSourceCapabilities::addFacingMode):
(WebCore::RealtimeMediaSourceCapabilities::encode const): Deleted.
(WebCore::RealtimeMediaSourceCapabilities::decode): Deleted.
* Source/WebCore/platform/mediastream/RealtimeMediaSourceSettings.cpp:
(WebCore::RealtimeMediaSourceSettings::videoFacingModeEnum):
* Source/WebCore/platform/mediastream/RealtimeMediaSourceSettings.h:
(WebCore::RealtimeMediaSourceSettings::RealtimeMediaSourceSettings):
(WTF::LogArgument&lt;WebCore::VideoFacingMode&gt;::toString):
(WebCore::RealtimeMediaSourceSettings::encode const): Deleted.
(WebCore::RealtimeMediaSourceSettings::decode): Deleted.
(WTF::LogArgument&lt;WebCore::RealtimeMediaSourceSettings::VideoFacingMode&gt;::toString): Deleted.
* Source/WebCore/platform/mediastream/RealtimeMediaSourceSupportedConstraints.h:
(WebCore::RealtimeMediaSourceSupportedConstraints::RealtimeMediaSourceSupportedConstraints):
(WebCore::RealtimeMediaSourceSupportedConstraints::encode const): Deleted.
(WebCore::RealtimeMediaSourceSupportedConstraints::decode): Deleted.
* Source/WebCore/platform/mediastream/VideoPreset.h:
(WebCore::FrameRateRange::encode const): Deleted.
(WebCore::FrameRateRange::decode): Deleted.
(WebCore::VideoPresetData::encode const): Deleted.
(WebCore::VideoPresetData::decode): Deleted.
* Source/WebCore/platform/mediastream/mac/AVVideoCaptureSource.mm:
(WebCore::AVVideoCaptureSource::settings):
(WebCore::AVVideoCaptureSource::capabilities):
* Source/WebCore/platform/mock/MockRealtimeVideoSource.cpp:
(WebCore::MockRealtimeVideoSource::drawText):
* Source/WebKit/Scripts/webkit/messages.py:
(serialized_identifiers):
* Source/WebKit/Scripts/webkit/tests/MessageArgumentDescriptions.cpp:
(IPC::serializedIdentifiers):
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:

Canonical link: <a href="https://commits.webkit.org/260561@main">https://commits.webkit.org/260561@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/56617e86640e636c518e22270a990fc05f139074

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/108651 "4 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/17752 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/41505 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/117761 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/117962 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/112534 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/19203 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/9029 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/100884 "Built successfully") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/114418 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/14398 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/97632 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/42374 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/5/builds/112181 "Passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/96373 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/29270 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/84106 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/10560 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/30620 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/11318 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/7534 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/16709 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/50216 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/7302 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/12906 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

youennf commented on Source/WebCore/platform/mediastream/RealtimeMediaSourceSettings.h:
```diff
@@ -35,21 +35,21 @@
 namespace WebCore {
 
+enum class VideoFacingMode : uint8_t { Unknown, User, Environment, Left, Right };
+enum class DisplaySurfaceType : uint8_t {
```
Preexisting but we should probably have a consistent style between the two enums (either one line or one enum per line).

youennf commented on Source/WebCore/platform/mediastream/RealtimeMediaSourceSettings.h:
```diff
@@ -72,6 +72,24 @@ class RealtimeMediaSourceSettings {
 
     explicit RealtimeMediaSourceSettings() = default;
+    RealtimeMediaSourceSettings(uint32_t width, uint32_t height, float aspectRatio, float frameRate, VideoFacingMode facingMode, double volume, uint32_t sampleRate, uint32_t sampleSize, bool echoCancellation, AtomString deviceId, AtomString groupId, AtomString label, DisplaySurfaceType displaySurface, bool logicalSurface, RealtimeMediaSourceSupportedConstraints supportedConstraints)
+        : m_width(WTFMove(width))
```
Moving a width is probably not needed, let's move deviceId, groupId, label and supportedConstraints.

youennf commented on Source/WebCore/platform/mediastream/RealtimeMediaSourceSettings.h:
```diff
@@ -72,6 +72,24 @@ class RealtimeMediaSourceSettings {
     WEBCORE_EXPORT OptionSet<RealtimeMediaSourceSettings::Flag> difference(const RealtimeMediaSourceSettings&) const;
 
     explicit RealtimeMediaSourceSettings() = default;
+    RealtimeMediaSourceSettings(uint32_t width, uint32_t height, float aspectRatio, float frameRate, VideoFacingMode facingMode, double volume, uint32_t sampleRate, uint32_t sampleSize, bool echoCancellation, AtomString deviceId, AtomString groupId, AtomString label, DisplaySurfaceType displaySurface, bool logicalSurface, RealtimeMediaSourceSupportedConstraints supportedConstraints)
```
AtomString&&, RealtimeMediaSourceSupportedConstraints&&

gavin-apple committed:
Port the remaining mediastream related types to the new IPC serialization format
https://bugs.webkit.org/show_bug.cgi?id=251860
rdar://105132585

Reviewed by Youenn Fablet.

This change ports the remaining MediaStream related types. This includes:
    - MediaConstraint::DataType
    - MediaConstraintType
    - MediaConstraint
    - IntConstraint
    - DoubleConstraint
    - BooleanConstraint
    - StringConstraint
    - FrameRateRange
    - VideoPresetData
    - RealtimeMediaSourceSupportedConstraints
    - VideoFacingMode
    - DisplaySurfaceType
    - RealtimeMediaSourceSettings
    - CaptureDevice::DeviceType
    - CaptureDevice
    - RealtimeMediaSourceCapabilities::EchoCancellation
    - RTCDataChannelInit
    - RTCDataChannelIdentifier

* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/platform/mediastream/CaptureDevice.h:
(WebCore::CaptureDevice::encode const): Deleted.
(WebCore::CaptureDevice::decode): Deleted.
* Source/WebCore/platform/mediastream/MediaConstraints.h:
(WebCore::NumericConstraint::NumericConstraint):
(WebCore::StringConstraint::StringConstraint):
(WebCore::MediaConstraint::encode const): Deleted.
(WebCore::MediaConstraint::decode): Deleted.
(WebCore::NumericConstraint::encode const): Deleted.
(WebCore::NumericConstraint::decode): Deleted.
(WebCore::StringConstraint::encode const): Deleted.
(WebCore::StringConstraint::decode): Deleted.
* Source/WebCore/platform/mediastream/RTCDataChannelHandler.h:
(WebCore::RTCDataChannelInit::encode const): Deleted.
(WebCore::RTCDataChannelInit::decode): Deleted.
* Source/WebCore/platform/mediastream/RTCDataChannelIdentifier.h:
(): Deleted.
(WebCore::RTCDataChannelIdentifier::encode const): Deleted.
(WebCore::RTCDataChannelIdentifier::decode): Deleted.
* Source/WebCore/platform/mediastream/RTCDataChannelLocalIdentifier.h: Copied from Source/WebCore/platform/mediastream/RTCDataChannelIdentifier.h.
* Source/WebCore/platform/mediastream/RealtimeMediaSource.h:
* Source/WebCore/platform/mediastream/RealtimeMediaSourceCapabilities.h:
(WebCore::RealtimeMediaSourceCapabilities::RealtimeMediaSourceCapabilities):
(WebCore::RealtimeMediaSourceCapabilities::facingMode const):
(WebCore::RealtimeMediaSourceCapabilities::addFacingMode):
(WebCore::RealtimeMediaSourceCapabilities::encode const): Deleted.
(WebCore::RealtimeMediaSourceCapabilities::decode): Deleted.
* Source/WebCore/platform/mediastream/RealtimeMediaSourceSettings.cpp:
(WebCore::RealtimeMediaSourceSettings::videoFacingModeEnum):
* Source/WebCore/platform/mediastream/RealtimeMediaSourceSettings.h:
(WebCore::RealtimeMediaSourceSettings::RealtimeMediaSourceSettings):
(WTF::LogArgument<WebCore::VideoFacingMode>::toString):
(WebCore::RealtimeMediaSourceSettings::encode const): Deleted.
(WebCore::RealtimeMediaSourceSettings::decode): Deleted.
(WTF::LogArgument<WebCore::RealtimeMediaSourceSettings::VideoFacingMode>::toString): Deleted.
* Source/WebCore/platform/mediastream/RealtimeMediaSourceSupportedConstraints.h:
(WebCore::RealtimeMediaSourceSupportedConstraints::RealtimeMediaSourceSupportedConstraints):
(WebCore::RealtimeMediaSourceSupportedConstraints::encode const): Deleted.
(WebCore::RealtimeMediaSourceSupportedConstraints::decode): Deleted.
* Source/WebCore/platform/mediastream/VideoPreset.h:
(WebCore::FrameRateRange::encode const): Deleted.
(WebCore::FrameRateRange::decode): Deleted.
(WebCore::VideoPresetData::encode const): Deleted.
(WebCore::VideoPresetData::decode): Deleted.
* Source/WebCore/platform/mediastream/mac/AVVideoCaptureSource.mm:
(WebCore::AVVideoCaptureSource::settings):
(WebCore::AVVideoCaptureSource::capabilities):
* Source/WebCore/platform/mock/MockRealtimeVideoSource.cpp:
(WebCore::MockRealtimeVideoSource::drawText):
* Source/WebKit/Scripts/webkit/messages.py:
(serialized_identifiers):
* Source/WebKit/Scripts/webkit/tests/MessageArgumentDescriptions.cpp:
(IPC::serializedIdentifiers):
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:

Canonical link: https://commits.webkit.org/260561@main
